packages/engineer/src/feature/dev-server.dev-server.env.ts

```diff
@@ -208,27 +216,25 @@ devServerFeature.setup(
             });
 
             const compilationPromises: Promise<void>[] = [];
-            for (const childCompiler of compiler.compilers) {
-                const devMiddleware = webpackDevMiddleware(childCompiler);
+
+            if (compiler.compilers.length > 0) {
+                const devMiddleware = webpackDevMiddleware(compiler);
                 disposables.add(
                     () => new Promise<void>((res) => devMiddleware.close(res))
                 );
                 app.use(devMiddleware);
-
-                if (webpackHot) {
-                    const hotMiddleware = webpackHotMiddleware(childCompiler);
-                    disposables.add(hotMiddleware.close);
-                    app.use(hotMiddleware);
-                }
                 compilationPromises.push(
                     new Promise<void>((resolve) => {
-                        childCompiler.hooks.done.tap('compiled', () => resolve());
+                        devMiddleware.waitUntilValid(() => resolve());
                     })
                 );
+                if (webpackHot) {
```

Comment:
do you need the none hot one when there is a hotMiddleware ?

Reply:
Yes, the hot middleware is responsible for the notification layer, not for the actual compilation management
Se example 
https://github.com/webpack-contrib/webpack-hot-middleware/blob/master/example/server.js#L24

From their docs
This module is only concerned with the mechanisms to connect a browser client to a webpack server & receive updates. It will subscribe to changes from the server and execute those changes using webpack's HMR API.